libs/cart/src/resolvers/empty-cart/empty-cart-resolver.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { ActionsSubject } from '@ngrx/store';
+import { filter, map } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+import { DaffCartActionTypes, DaffCartLoadSuccess } from '../../actions/public_api';
+
+/**
+ * This "resolver" isn't strictly an "angular resolver", because Daffodil isn't meant to touch actual routes for an app.
+ * Instead, this service should be used in conjunction with an angular routing resolver to guard against routing to a 
+ * certain page (usually the checkout page) when the cart is empty.
+ */
+@Injectable()
+export class DaffEmptyCartResolver {
+  constructor(private dispatcher: ActionsSubject) {}
+
+  isCartEmpty(): Observable<boolean> {
+    return this.dispatcher.pipe(
```

Comment:
Isn't this logic already built into selectors? Couldn't we just inject the facade here and use the existing `isCartEmpty` selector?

Reply:
yes